### DISCUSSION TITLE
用Cloudflare Workers建立一个免费的词典笔用New Bing代理

### DISCUSSION THREAD

**Discussion by socketcat** [PenUniverse/PenMods#209] [奇技淫巧]
Workers自带的域名被墙阻断了，无法直连，所以我们需要获取一个域名
可以上[哪煮米](https://www.nazhumi.com/)查看哪里购买便宜域名，或者也可以申请一个免费域名，这里以ClouDNS为例：
1. 进入[官网](https://www.cloudns.net/)创建域名，这一步比较简单，如有不懂可看[这篇文章](https://blog.misaka.rest/2023/12/17/cloudns-free-domain/)。注：目前ClouDNS风控比较严，请选用大厂邮箱和干净IP注册，如没有干净IP可以使用Cloudflare WARP（一个免费vpn服务）进行注册。
2. 在Cloudflare添加你的域名，并在ClouDNS添加NS解析。
3. 在Cloudflare面板侧边栏中选中Workers & Pages，点击创建应用，选择创建 Worker，随意取一个名字后点击部署。
4. 创建后点击Edit code，会打开一个网页编辑器。将原先的内容删除，替换为[这里的内容](https://raw.githubusercontent.com/Harry-zklcdc/go-proxy-bingai/master/cloudflare/worker.js)。然后点击左侧栏的文件图标，再创建两个新文件：[bjs.js](https://raw.githubusercontent.com/Harry-zklcdc/go-proxy-bingai/master/cloudflare/bjs.js)和[bingapi.js](https://raw.githubusercontent.com/Harry-zklcdc/go-proxy-bingai/master/cloudflare/bingapi.js)，内容即为链接里文件内容。编辑完成后，点击右上角Save and deploy按钮。
5. 之后返回Workers界面，点上面的Settings，再点Triggers，添加你的自定义域名。
6. 再次进入ClouDNS控制面板，为你刚刚添加到workers的子域名添加一个A解析：104.18.8.215，再添加一个AAAA解析：2606:4700:9a6c:5494:3e85:5045:5bad:33e6
7. 到这里代理就创建完毕了，Signature申请地址为https://域名/turing/conversation/create，ChatHub地址为wss://域名/sydney/chathub/，填入词典笔就可以用了。
8. 如果你是1.2.0版本，Signature申请地址需要多加一个参数，地址为https://域名/turing/conversation/create?bundleVersion=1.0

**Comment by dudu-Dev0** [comment]
workers里全是报错保存不了怎么办![Screenshot_2024-03-19-01-23-52-240_com.android.chrome.jpg](https://github.com/PenUniverse/PenMods-release/assets/86507157/5e369a05-603e-496f-ad99-24d02b7146cd)

**Reply by SkySight-666** [reply]
佬能把你复制到的bjs.js发一份吗，楼主的链接寄了，谢谢😊

**Comment by laoaideqi** [comment]
![QQ图片20240427003347](https://github.com/PenUniverse/PenMods-release/assets/99399450/382fa315-f4d4-41ab-bf90-c239b27da76f)
这个怎么办

**Comment by SkySight-666** [comment]
bjs.js进去变404了

**Reply by Chengzi600** [reply]
[https://github.com/Harry-zklcdc/go-proxy-bingai/tree/98fb83fe18d0ee4af034e2d4a23866cb63103a3a/cloudflare](https://github.com/Harry-zklcdc/go-proxy-bingai/tree/98fb83fe18d0ee4af034e2d4a23866cb63103a3a/cloudflare)

**Reply by SkySight-666** [reply]
> https://github.com/Harry-zklcdc/go-proxy-bingai/tree/98fb83fe18d0ee4af034e2d4a23866cb63103a3a/cloudflare

啊 是搭建成功了 但是好像不生效（词典笔提问完就没报错也没反应了 ，wss测试连接正常，penmods1.30 （疑惑
![image](https://github.com/PenUniverse/PenMods-release/assets/110458588/bd26b628-055f-4ab6-b8e1-976d103a66c0)

**Reply by Zack88604** [reply]
> > https://github.com/Harry-zklcdc/go-proxy-bingai/tree/98fb83fe18d0ee4af034e2d4a23866cb63103a3a/cloudflare
> 
> 啊 是搭建成功了 但是好像不生效（词典笔提问完就没报错也没反应了 ，wss测试连接正常，penmods1.30 （疑惑 ![image](https://private-user-images.githubusercontent.com/110458588/342780205-bd26b628-055f-4ab6-b8e1-976d103a66c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2NzQ3OTEsIm5iZiI6MTcyMDY3NDQ5MSwicGF0aCI6Ii8xMTA0NTg1ODgvMzQyNzgwMjA1LWJkMjZiNjI4LTA1NWYtNGFiNi1iOGUxLTk3NmQxMDNhNjZjMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxMVQwNTA4MTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YzhjNjZmZTZmZDhhMjY5NjFjNjJiZTQ5MmQxMzA2MzdlNWE4MjZjNzMyNDZmNTc0ZmZjNDBmYzcyNWQxZmUxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.f-c161NKG0LbAu4sO-VPJf8swPwMBFdfjUJhym-Z720)

对于CF的worker 很多时候似乎需要冷启动（好像和CF的宣传不太一致
发送请求以后没反应很正常 耐心等待一会应该会有回传（或者报错

**Comment by tofuture-1** [comment]
> bjs.js进去变404了

可以分享一下你的搭建流程吗，目前卡在worker文件开头的引用报错

**Reply by SkySight-666** [reply]
就是直接粘贴代码啊💦
可以给我看一下你的worker吗

**Reply by Zack88604** [reply]
可以参考[这里](https://www.b1ng.chat)的教程，cloudflare pages & workers可用

**Reply by SkySight-666** [reply]
> 可以参考[这里](https://www.b1ng.chat)的教程，cloudflare pages & workers可用

新版的b1ng AI自带**反代**吗 如果是这样的话，我就更新了

**Reply by tofuture-1** [reply]
> 就是直接粘贴代码啊💦
> 可以给我看一下你的worker吗

搞定了，原帖链接的woker.js与可用的不一样，用前面几楼分享出来的原项目中的三个文件就正常了
